### PR TITLE
Multithread application

### PR DESCRIPTION
**Pull request by MihailRom**
Multithreaded application including tests

Issue #278

### CODE REVIEW

File: HW8/Mikhail_Romashov/Multithread/Multithread.Core/Services/LinkService.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+
+        public LinkEntity GetLinkById(int id)
+        {
+            if (!this.linkTableRepository.ContainsById(id))
+                throw new ArgumentException($"Can`t find item by this id = {id}");
+            return this.linkTableRepository.GetById(id);
+        }
+      
+        public LinkEntity GetTraderById(int linkId)
```
Comment (chesdenis) [inline HW8/Mikhail_Romashov/Multithread/Multithread.Core/Services/LinkService.cs]: :) ?

Comment (chesdenis) [reply HW8/Mikhail_Romashov/Multithread/Multithread.Core/Services/LinkService.cs]: GetTrader...??

File: HW8/Mikhail_Romashov/Multithread/MultithreadConsoleApp.Tests/HtmlReaderTests.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        IHtmlReader htmlReader;
+        HttpClient httpClient;
+        [TestInitialize]
+        public void Initialize()
+        {
+            HttpMessageHandler myHttpMessageHandler = new MyHandler();
+            httpClient = new HttpClient(myHttpMessageHandler);
+            htmlReader = new HtmlReader(httpClient);
+        }
+        public class MyHandler : HttpMessageHandler
```
Comment (chesdenis) [inline HW8/Mikhail_Romashov/Multithread/MultithreadConsoleApp.Tests/HtmlReaderTests.cs]: Please never use such names... if this is mock or stub, then it should be named like FakeHttpHandler, or MockHttpHandler.

File: HW8/Mikhail_Romashov/Multithread/MultithreadConsoleApp/Components/IOFile.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.IO;
+
+namespace MultithreadConsoleApp.Components
+{
+    public class IOFile
```
Comment (chesdenis) [inline HW8/Mikhail_Romashov/Multithread/MultithreadConsoleApp/Components/IOFile.cs]: Incorrect name, can be FileSystemManager

File: HW8/Mikhail_Romashov/Multithread/MultithreadConsoleApp.Tests/HtmlReaderTests.cs (hunk 2 above)
Comment (MihailRom) [reply HW8/Mikhail_Romashov/Multithread/MultithreadConsoleApp.Tests/HtmlReaderTests.cs]: Oh, stupid mistake. Fixed
Tkank

File: HW8/Mikhail_Romashov/Multithread/Multithread.Core/Services/LinkService.cs (hunk 1 above)
Comment (MihailRom) [reply HW8/Mikhail_Romashov/Multithread/Multithread.Core/Services/LinkService.cs]: Extra method, deleted

**Pull request merged**